oas3.yaml

```diff
@@ -0,0 +1,328 @@
+openapi: 3.0.1
+info:
+  title: Shelter API
+  description: |
+    A website which provides information about pets which are being sheltered and are available
+    for adoption.
+
+  license:
+    name: MIT
+  version: 1.0.0
+
+servers:
+  - url: http://localhost:5000
+  - url: https://localhost:5001
+
+tags:
+  - name: health
+    description: APIs used to determine whether your service is healthy and running as expected.
+
+  - name: pets
+    description: APIs used to enumerate and manage the collection of pets which are available for adoption.
+
+paths:
+  /api/v1/health:
+    get:
+      tags:
+        - health
+
+      summary: Get Health (v1)
+      description: Gets information about the current health state of your service.
+      operationId: health_v1
+      responses:
+        200:
+          description: The service is healthy.
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/HealthV1"
+              example:
+                status: Healthy
+                results:
+                  database:
+                    status: Healthy
+                    description: Connected to upstream database server.
+                    data: {}
+        500:
+          description: The service is unhealthy.
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/HealthV1"
+              example:
+                status: Unhealthy
+                results:
+                  database:
+                    status: Unhealthy
+                    description: Unable to connect to the upstream database server.
+                    data: {}
+
+  /api/v1/pets:
+    get:
+      tags:
+        - pets
+      summary: Get Pets (v1)
+      description: Gets the list of pets which have been registered with our shelter.
+      operationId: pets_list_v1
+      responses:
+        200:
+          description: The list of pets.
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: "#/components/schemas/PetV1"
+        500:
+          description: An error occurred on the server.
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Error"
+
+    post:
```

Comment:
Also consider a bad request response for create, where/if e.g. the caller didn't specify required values or the pet json is not per schema.  Same for modify below.